File: lib/py/etl/paradicms_etl/models/cms/cms_organization.py
from typing import Optional

from rdflib import URIRef, Graph, BNode, FOAF

from paradicms_etl.models.cms.cms_agent import CmsAgent
from paradicms_etl.models.organization import Organization


class CmsOrganization(CmsAgent, Organization):
    class Builder(CmsAgent.Builder):
        def add_page(self, page: URIRef) -> "CmsOrganization.Builder":
            super().add_page(page)
            return self

        def build(self):
            return CmsOrganization(self._resource)

    @classmethod
    def builder(cls, *, name: str, uri: Optional[URIRef] = None) -> Builder:
        builder = cls.Builder(Graph().resource(uri if uri is not None else BNode()))
        builder.set(FOAF.name, name)
        return builder

File: lib/ts/models/src/CmsOrganization.ts
import { AgentVisitor } from "./AgentVisitor";
import { CmsAgent } from "./CmsAgent";
import { Organization } from "./Organization";

export class CmsOrganization extends CmsAgent implements Organization {
  override accept<T>(visitor: AgentVisitor<T>): T {
    return visitor.visitOrganization(this);
  }    
}
